includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -309,7 +309,7 @@ protected function read_visibility( &$product ) {
 	protected function read_attributes( &$product ) {
 		$meta_values = maybe_unserialize( get_post_meta( $product->get_id(), '_product_attributes', true ) );
 
-		if ( $meta_values ) {
+		if ( $meta_values && !empty( $meta_values ) ) {
```

Comment:
I have pushed my updated code which depends on your suggestion for #14089 . My pull request is  #14099 . Please let me know if there are any changes.